lms/djangoapps/django_comment_client/base/views.py

```diff
@@ -481,9 +524,51 @@ def follow_thread(request, course_id, thread_id):
     user = cc.User.from_django_user(request.user)
     thread = cc.Thread.find(thread_id)
     user.follow(thread)
+    # Feature Flag to check that notifications are enabled or not.
+    if settings.FEATURES["NOTIFICATIONS_ENABLED"]:
+        # only send notifications when the user
+        # who is following the thread is not the same
+        # who created the thread
+        original_poster_id = int(thread.user_id)
+        action_user_id = int(request.user.id)
+        if not original_poster_id == action_user_id:
+            register_publish_notifications(
+                msg_type_name='open-edx.lms.discussions.thread-followed',
+                course_id=course_id,
+                original_poster_id=original_poster_id,
+                action_user_id=action_user_id,
+                action_username=request.user.username,
+                thread_title=thread.title,
+                link_to_thread=permalink(thread)
+            )
     return JsonResponse({})
 
 
+def register_publish_notifications(msg_type_name, course_id, original_poster_id, action_user_id,
```

Comment:
@chrisndodge this exceeds the max function name length. pylint/pep8 gives error invalid-name